task8/TutterApp/src/main/java/dao/PostDAOImpl.java

```diff
@@ -0,0 +1,112 @@
+package dao;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import forms.FilterForm;
+import models.Constants;
+import models.Post;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PostDAOImpl implements PostDAO {
+    private static List<Post> posts;
+
+    public PostDAOImpl() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.setDateFormat(new SimpleDateFormat(Constants.DATE_FORMAT));
+
+        try {
+            posts = objectMapper.readValue(new File(Constants.PATH), new TypeReference<>() {
```

Comment:
Can't figure out what is the use of an empty `TypeReference` override. Is it actually working?